### DISCUSSION TITLE
[Feature] Ability to change timestamps more precisely

### DISCUSSION THREAD

**Discussion by ajpanton** [immich-app/immich#11335] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Editing a photo's timestamp is only possible at the minute level. This makes it difficult to correctly order photos taken very close to each other. Even worse, if you only wanted to change the timezone and didn't even mean to touch the timestamp, the seconds still get reset to zero.

The resulting XMP file seems to already support millisecond level precision. You might not actually need that, but still, if the original photo already has millisecond precision, it's unnecessary to change that if all you want is to set a timezone or change the day. The input window already handily selects the part of the time you click on, making it easy to change only the part that you want, so I wouldn't even mind if it included both seconds and milliseconds, at least in the browser version. Editing metadata is already not really everyday usage, so if the user goes to actually edit the metadata, restricting it seems unnecessary.

### Platform

- [X] Server
- [X] Web
- [ ] Mobile

**Comment by trommegutten** [comment]
I fully agree with this request. It’s especially problematic that editing timestamps via the web interface resets the seconds (and milliseconds) even if you only intended to adjust the date. This unintentionally corrupts precise timecode information, which is critical for correct photo ordering, especially in workflows involving high burst rates or synchronized media from multiple devices. Preserving full timestamp precision during edits should be considered essential functionality.
